### PR TITLE
Merge preview into main

### PR DESCRIPTION
**Pull request by rossgrambo**
## Why this PR?

Preparing preview content for release by merging into main.

Will be merged after: https://github.com/microsoft/FeatureManagement-Dotnet/pull/493

## Visible Changes

* Variants & IVariantFeatureManager
* Telemetry
  * ActivitySource Microsoft.FeatureManagement & EvaluationEvent as ActivityEvent
  * TargetingHttpContextMiddleware & TargetingTelemetryInitializer
  * Application Insights Event Publisher
  * .AddApplicationInsights builder extension
* VariantService
* .WithTargeting adds HttpContextAccessor if it's not already added

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -9,7 +9,7 @@ Feature management provides a way to develop and expose application functionalit
 
 [**Quickstart**](https://learn.microsoft.com/azure/azure-app-configuration/quickstart-feature-flag-dotnet): A quickstart guide is available to learn how to integrate feature flags from *Azure App Configuration* into your .NET applications.
 
-[**Feature Reference**](https://learn.microsoft.com/azure/azure-app-configuration/feature-management-dotnet-reference): This document provides a full feature rundown.
+[**Feature Reference**](https://learn.microsoft.com/azure/azure-app-configuration/feature-management-dotnet-reference?pivots=preview-version): This document provides a full feature rundown.
```
Comment (zhiyuanliang-ms) [inline README.md:12]: This change should be removed after #493

File: src/Microsoft.FeatureManagement.AspNetCore/AspNetCoreFeatureManagementBuilderExtensions.cs [hunk 2]
```diff
@@ -1,6 +1,8 @@
 ﻿// Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
 //
+using Microsoft.AspNetCore.Builder;
```
Comment (zhiyuanliang-ms) [inline src/Microsoft.FeatureManagement.AspNetCore/AspNetCoreFeatureManagementBuilderExtensions.cs:4]: This seems to be not used. We should remove it.

File: src/Microsoft.FeatureManagement/FeatureManagementOptions.cs [hunk 3]
```diff
@@ -1,6 +1,10 @@
 ﻿// Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
 //
+using Microsoft.FeatureManagement.Telemetry;
```
Comment (zhiyuanliang-ms) [inline src/Microsoft.FeatureManagement/FeatureManagementOptions.cs:4]: These imports are not used. We should remove them.

File: src/Microsoft.FeatureManagement/FeatureStatus.cs [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    /// <summary>
+    /// Describes how a feature's state will be evaluated.
+    /// </summary>
+    public enum FeatureStatus
+    {
+        /// <summary>
+        /// The state of the feature is conditional upon the feature evaluation pipeline.
+        /// </summary>
+        Conditional,
+        /// <summary>
```
Comment (zhiyuanliang-ms) [inline src/Microsoft.FeatureManagement/FeatureStatus.cs:15]: Could you add an empty line here?

File: src/Microsoft.FeatureManagement/StatusOverride.cs [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+{
+    /// <summary>
+    /// Provides the capability to override whether a feature is considered enabled or disabled when a variant is assigned.
+    /// </summary>
+    public enum StatusOverride
+    {
+        /// <summary>
+        /// Does not affect the feature state.
+        /// </summary>
+        None,
```
Comment (zhiyuanliang-ms) [inline src/Microsoft.FeatureManagement/StatusOverride.cs:14]: Could you add empty lines between enum values? Just like what we've done in VariantAssignmentReason.cs

**Review by zhiyuanliang-ms — APPROVED** [review]

**Comment by rossgrambo** [comment]
Rerun fixed the issue

**Pull request merged**